### ISSUE TITLE
New drive not recognised until later on.

### ISSUE THREAD

**Issue by woyteck1** [adriankumpf/teslamate#2450]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

So, a new driving is reported as driving but from nowhere to nowwhere and doesn't include cutoff time.
Then another entry appears when driving got recognized, but from the, say, midpoint of the drive.
This how it looks in timeline:
![image](https://user-images.githubusercontent.com/25311523/152990394-1195be9e-a5a6-4ccf-aa67-e4e5e9bce541.png)
And this how it looks in drives:
![image](https://user-images.githubusercontent.com/25311523/152990861-41ed7b7c-8410-4a0d-9da5-17f9894b27e8.png)

Then, this faulty timeline entry doesn't have end time, so when clicked on, it displays data until now.
The usage count is N/A and it's very odd.
![image](https://user-images.githubusercontent.com/25311523/152991169-06afdc6c-d6bf-48bc-92af-5f53983c8cd5.png)

### Expected Behavior

Expected would be to register it as a drive from the beginning to the end, or at least up to the next drive entry in the DB.

### Steps To Reproduce

It happened two or three times to me so far.
I am unable to reproduce it as it was at different conditions each time.

### Relevant log output

```shell
DOnt have logs from the event as I updated the teslamate afterwards.
```

### Type of installation

Docker

### Version

v1.26.0 and v1.26.1

**Comment by cwanja** [comment]
You should still be able to capture the logs regardless of updating. Container logs are still there. 

If you click on the timestamp from the `Timeline` dashboard and it opens a trip - are there any drives listed there? It looks like there was a cellular signal interruption and the drive was "stopped" due to a lack of signal. And then picked back up again. 

If we can identify the drive ID, you might need to terminate the drive manually: https://docs.teslamate.org/docs/maintenance/manually_fixing_data#terminate-a-drive-or-charge

**Comment by woyteck1** [comment]
Yes, there are multiple drives listed. 
Can I close the drive, but give it a specific time? I would still like to keep the data from it, but only for actual time of the drive and not until now (which is over a week now).

**Comment by cwanja** [comment]
You have to provide the terminate drive script a drive ID. Which you should be able to get from the trip dashboard after clicking on the time stamp. 

The data should all be intact after terminating the drive

**Comment by woyteck1** [comment]
I did get this:

```
# docker-compose exec teslamate bin/teslamate rpc \
>  "TeslaMate.Repo.get!(TeslaMate.Log.Drive, 313) |> TeslaMate.Log.close_drive()"
bash: !: event not found

```
Didn't work.

**Comment by init6-git** [comment]
I've got the same. Only happens on long drives (one specific route, but that's the only long drive we do)
<img width="1201" alt="Screenshot 2022-02-12 at 12 02 53" src="https://user-images.githubusercontent.com/85936842/153710788-d0e0c4a4-04c4-4477-80c4-c0dfbdf73604.png">

Seems to split in different locations. Here's the logs for the latest drive:
`teslamate_1           | 2022-02-11 10:04:01.046 [info] GET [https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data)
[_data](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data) -> 408 (9407.290 ms)
teslamate_1           | 2022-02-11 10:04:01.047 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_descripti
on" => "", "response" => nil}
teslamate_1           | 2022-02-11 10:04:01.047 car_id=1 [error] Error / :unknown
teslamate_1           | 2022-02-11 10:04:30.515 [info] GET [https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data)
[_data](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data) -> 408 (9467.406 ms)
teslamate_1           | 2022-02-11 10:04:30.516 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_descripti
on" => "", "response" => nil}
teslamate_1           | 2022-02-11 10:04:30.516 car_id=1 [error] Error / :unknown
teslamate_1           | 2022-02-11 10:04:59.984 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle
_data -> 408 (9467.010 ms)
teslamate_1           | 2022-02-11 10:04:59.984 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_descripti
on" => "", "response" => nil}
teslamate_1           | 2022-02-11 10:04:59.985 car_id=1 [error] Error / :unknown
teslamate_1           | 2022-02-11 10:05:29.412 [info] GET https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle
_data -> 408 (9426.273 ms)
teslamate_1           | 2022-02-11 10:05:29.412 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_descripti
on" => "", "response" => nil}
teslamate_1           | 2022-02-11 10:05:29.413 car_id=1 [error] Error / :unknown
teslamate_1           | 2022-02-11 10:05:51.207 car_id=1 [info] Suspending logging
teslamate_1           | 2022-02-11 10:09:39.285 [notice]     :alarm_handler: {:clear, TeslaMate.Vehicles.Vehicle_1_api_error}
teslamate_1           | 2022-02-11 10:19:31.980 [info] Start of drive initiated by: %TeslaApi.Stream.Data{elevation: 50, est_heading
: 46, est_lat: 55.819882, est_lng: -4.208232, est_range: 195, heading: 52, odometer: 9880.8, power: 5, range: 309, shift_state: "D",
 soc: 96, speed: 1, time: ~U[2022-02-11 10:19:29.942Z]}
teslamate_1           | 2022-02-11 10:19:31.981 car_id=1 [info] Driving / Start
teslamate_1           | 2022-02-11 10:29:48.024 [info] Refreshing access token ...
teslamate_1           | 2022-02-11 10:29:48.388 [info] POST https://auth.tesla.com/oauth2/v3/token -> 200 (363.531 ms)
teslamate_1           | 2022-02-11 10:29:48.395 [info] Scheduling token refresh in 6 h
teslamate_1           | 2022-02-11 10:58:11.184 [info] GET /
teslamate_1           | 2022-02-11 10:58:11.189 [info] Sent 200 in 4ms
teslamate_1           | 2022-02-11 11:58:11.205 [info] GET /
teslamate_1           | 2022-02-11 11:58:11.214 [info] Sent 200 in 8ms
teslamate_1           | 2022-02-11 11:58:29.012 [info] GET [https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data)
[_data](https://owner-api.teslamotors.com/api/1/vehicles/1493098064281846/vehicle_data) -> 408 (9437.089 ms)
teslamate_1           | 2022-02-11 11:58:29.012 [warning] TeslaApi.Error / %{"error" => "{\"error\": \"timeout\"}", "error_descripti
on" => "", "response" => nil}
teslamate_1           | 2022-02-11 11:58:29.013 car_id=1 [error] Error / :unknown
teslamate_1           | 2022-02-11 12:34:18.005 [error] ** (RuntimeError) Client Error: %{"error_type" => "client_error", "msg_type"
 => "data:error", "tag" => "165879258103", "value" => "Can't validate token. "}
teslamate_1           |     (teslamate 1.26.1) lib/tesla_api/stream.ex:165: TeslaApi.Stream.handle_frame/2
teslamate_1           |     (websockex 0.4.3) lib/websockex.ex:1113: WebSockex.try_callback/3
teslamate_1           |     (websockex 0.4.3) lib/websockex.ex:797: WebSockex.common_handle/4
teslamate_1           |     (stdlib 3.17) proc_lib.erl:226: :proc_lib.init_p_do_apply/3
teslamate_1           | 
teslamate_1           | 2022-02-11 12:34:20.865 car_id=1 [info] Start / :online
teslamate_1           | 2022-02-11 12:34:20.913 car_id=1 [info] Connecting ...
teslamate_1           | 2022-02-11 12:34:20.914 [info] Start of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 1644582
858, heading: 45, latitude: 57.03197, longitude: -2.158485, native_latitude: 57.03197, native_location_supported: 1, native_longitud
e: -2.158485, native_type: "wgs", power: -4, shift_state: "D", speed: 79, timestamp: 1644582860459}
teslamate_1           | 2022-02-11 12:34:20.914 car_id=1 [info] Driving / Start
teslamate_1           | 2022-02-11 12:44:25.692 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (1026.873 ms)
teslamate_1           | 2022-02-11 12:44:26.567 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (127.004 ms)
teslamate_1           | 2022-02-11 12:44:26.738 [info] End of drive initiated by: %TeslaApi.Vehicle.State.Drive{gps_as_of: 164458345
6, heading: 111, latitude: 57.110185, longitude: -2.092592, native_latitude: 57.110185, native_location_supported: 1, native_longitu
de: -2.092592, native_type: "wgs", power: 1, shift_state: "P", speed: 0, timestamp: 1644583458602}
teslamate_1           | 2022-02-11 12:44:26.738 car_id=1 [info] Driving / Ended / 10 km – 10 min
teslamate_1           | 2022-02-11 12:44:26.738 car_id=1 [info] Start / :online
teslamate_1           | 2022-02-11 12:44:50.696 car_id=1 [info] Charging detected: -5 kW
teslamate_1           | 2022-02-11 12:44:50.952 car_id=1 [info] Charging detected: -14 kW
teslamate_1           | 2022-02-11 12:44:50.952 car_id=1 [info] Fetch already in progress ...
teslamate_1           | 2022-02-11 12:44:51.195 car_id=1 [info] Charging detected: -28 kW
teslamate_1           | 2022-02-11 12:44:51.195 car_id=1 [info] Fetch already in progress ...
teslamate_1           | 2022-02-11 12:44:51.650 [info] GET https://nominatim.openstreetmap.org/reverse -> 200 (344.427 ms)
teslamate_1           | 2022-02-11 12:44:52.352 car_id=1 [info] Charging / SOC: 20% / Aberdeen Supercharger`

**Comment by path3dev** [comment]
Just wanted to chip in and say that I had the same issue. But other than that, I noticed that especially after the car has been sleeping for a long time, then the first drive is never logged. Had this happen twice so far after the car had just been staying parked for 2 and 6 days respectively.
